pkg/connection/fake/mocks.go

```diff
@@ -28,61 +30,26 @@ import (
 	"github.com/crossplane/crossplane-runtime/pkg/connection/store"
 )
 
-// MockSecretOwner is a mock object that satisfies ConnectionSecretOwner
-// interface.
-type MockSecretOwner struct {
-	runtime.Object
-	metav1.ObjectMeta
-
-	To *v1.PublishConnectionDetailsTo
-}
-
-// GetPublishConnectionDetailsTo returns the publish connection details to reference.
-func (m *MockSecretOwner) GetPublishConnectionDetailsTo() *v1.PublishConnectionDetailsTo {
-	return m.To
-}
-
-// SetPublishConnectionDetailsTo sets the publish connection details to reference.
-func (m *MockSecretOwner) SetPublishConnectionDetailsTo(t *v1.PublishConnectionDetailsTo) {
-	m.To = t
-}
-
-// GetObjectKind returns schema.ObjectKind.
-func (m *MockSecretOwner) GetObjectKind() schema.ObjectKind {
-	return schema.EmptyObjectKind
-}
-
-// DeepCopyObject returns a copy of the object as runtime.Object
-func (m *MockSecretOwner) DeepCopyObject() runtime.Object {
-	out := &MockSecretOwner{}
-	j, err := json.Marshal(m)
-	if err != nil {
-		panic(err)
-	}
-	_ = json.Unmarshal(j, out)
-	return out
-}
-
 // SecretStore is a fake SecretStore
 type SecretStore struct {
-	ReadKeyValuesFn   func(ctx context.Context, i store.Secret) (store.KeyValues, error)
-	WriteKeyValuesFn  func(ctx context.Context, i store.Secret, kv store.KeyValues) error
-	DeleteKeyValuesFn func(ctx context.Context, i store.Secret, kv store.KeyValues) error
+	ReadKeyValuesFn   func(ctx context.Context, i store.Secret) (managed.ConnectionDetails, error)
+	WriteKeyValuesFn  func(ctx context.Context, i store.Secret, conn managed.ConnectionDetails) error
+	DeleteKeyValuesFn func(ctx context.Context, i store.Secret, conn managed.ConnectionDetails) error
```

Comment:
```suggestion
	DeleteConnectionDetailsFn func(ctx context.Context, i store.Secret, conn managed.ConnectionDetails) error
```

Is it worth changing the names of these methods to match the new args/return values?

Reply:
Hmm, checking this again, I didn't want to leak `managed.ConnectionDetails` in `connection.store` package because this is the interface that SecretStore implementations need to satisfy and these are just KeyValues for them, i.e. connection details does not mean something for a secret store.
Refactored accordingly.